lib/fastlane/plugin/wpmreleasetoolkit/actions/common/upload_to_s3.rb

```diff
@@ -22,11 +22,25 @@ def self.run(params)
 
         if file_is_already_uploaded?(bucket, key)
           message = "File already exists in S3 bucket #{bucket} at #{key}"
-          if params[:skip_if_exists]
+
+          # skip_if_exists is deprecated but to keep backward compatibility we still support it by reading it only if if_exists is not set.
+          if params[:if_exists].nil?
+            if params[:skip_if_exists]
+              UI.important("#{message}. Skipping upload.")
+              return key
+            else
+              UI.user_error!(message)
+            end
+          end
+
+          case params[:if_exists].to_sym
```

Comment:
I don't like that we need to "remember" to call `to_sym` here but there's no way to set the type as `Symbol`.

The docs suggest storing the value in a class property (instance [in the docs](https://docs.fastlane.tools/advanced/actions/#polymorphic-parameters), but we use `self.run`) but I don't know... we read everything else from `params` that it would seem odd to, again, have to "remember" to do that.

Reply:
[According to the source code it should be possible to set `type: Symbol` though](https://github.com/fastlane/fastlane/blob/master/fastlane_core/lib/fastlane_core/configuration/config_item.rb#L278).

I think what is misleading in the docs is that you thought you needed polymorphic parameters, to allow both `String` and `Symbol` to be passed.
But it happens that for some types like `Symbol` and `Array`, `ConfigItem` is able to auto-convert to the expected type when we pass just a `String` (I think this build-in auto-conversion was baked into `ConfigItem` mostly to make it easy to support calling actions directly from the Terminal via `fastlane run myaction`)